Null charset name

### Affected version

3.11.3

### Bug description

Happens on 3.11.3, works on 3.11.2, JDK 24-25:

```
2025-08-18T10:47:52.2774947Z [ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.11.3:jar (attach-javadocs) on project trino-jdbc: Execution attach-javadocs of goal org.apache.maven.plugins:maven-javadoc-plugin:3.11.3:jar failed: Null charset name -> [Help 1]
2025-08-18T10:47:52.2776055Z org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.11.3:jar (attach-javadocs) on project trino-jdbc: Execution attach-javadocs of goal org.apache.maven.plugins:maven-javadoc-plugin:3.11.3:jar failed: Null charset name
2025-08-18T10:47:52.2776367Z     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
2025-08-18T10:47:52.2776662Z     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
2025-08-18T10:47:52.2776949Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
2025-08-18T10:47:52.2777233Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
2025-08-18T10:47:52.2777524Z     at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
2025-08-18T10:47:52.2777798Z     at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
2025-08-18T10:47:52.2778187Z     at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
2025-08-18T10:47:52.2778468Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
2025-08-18T10:47:52.2778940Z     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
2025-08-18T10:47:52.2779261Z     at io.takari.maven.builder.smart.SmartBuilderImpl.buildProject (SmartBuilderImpl.java:206)
2025-08-18T10:47:52.2779593Z     at io.takari.maven.builder.smart.SmartBuilderImpl$ProjectBuildTask.run (SmartBuilderImpl.java:71)
2025-08-18T10:47:52.2779899Z     at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:545)
2025-08-18T10:47:52.2780151Z     at java.util.concurrent.FutureTask.run (FutureTask.java:328)
2025-08-18T10:47:52.2780436Z     at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1095)
2025-08-18T10:47:52.2780715Z     at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:619)
2025-08-18T10:47:52.2780855Z     at java.lang.Thread.run (Thread.java:1447)
2025-08-18T10:47:52.2781476Z Caused by: org.apache.maven.plugin.PluginExecutionException: Execution attach-javadocs of goal org.apache.maven.plugins:maven-javadoc-plugin:3.11.3:jar failed: Null charset name
2025-08-18T10:47:52.2781847Z     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:133)
2025-08-18T10:47:52.2782146Z     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
2025-08-18T10:47:52.2782493Z     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
2025-08-18T10:47:52.2782784Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
2025-08-18T10:47:52.2783065Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
2025-08-18T10:47:52.2783350Z     at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
2025-08-18T10:47:52.2783619Z     at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
2025-08-18T10:47:52.2784003Z     at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
2025-08-18T10:47:52.2784284Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
2025-08-18T10:47:52.2784688Z     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
2025-08-18T10:47:52.2785010Z     at io.takari.maven.builder.smart.SmartBuilderImpl.buildProject (SmartBuilderImpl.java:206)
2025-08-18T10:47:52.2785346Z     at io.takari.maven.builder.smart.SmartBuilderImpl$ProjectBuildTask.run (SmartBuilderImpl.java:71)
2025-08-18T10:47:52.2785598Z     at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:545)
2025-08-18T10:47:52.2785890Z     at java.util.concurrent.FutureTask.run (FutureTask.java:328)
2025-08-18T10:47:52.2786173Z     at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1095)
2025-08-18T10:47:52.2786445Z     at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:619)
2025-08-18T10:47:52.2786587Z     at java.lang.Thread.run (Thread.java:1447)
2025-08-18T10:47:52.2786789Z Caused by: java.lang.IllegalArgumentException: Null charset name
2025-08-18T10:47:52.2786962Z     at java.nio.charset.Charset.lookup (Charset.java:460)
2025-08-18T10:47:52.2787131Z     at java.nio.charset.Charset.forName (Charset.java:531)
2025-08-18T10:47:52.2787406Z     at org.apache.maven.plugins.javadoc.JavadocUtil.readFile (JavadocUtil.java:870)
2025-08-18T10:47:52.2787786Z     at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.getModulesLinks (AbstractJavadocMojo.java:5396)
2025-08-18T10:47:52.2788164Z     at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.getOfflineLinks (AbstractJavadocMojo.java:3767)
2025-08-18T10:47:52.2788526Z     at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.executeReport (AbstractJavadocMojo.java:1982)
2025-08-18T10:47:52.2788824Z     at org.apache.maven.plugins.javadoc.JavadocJarMojo.doExecute (JavadocJarMojo.java:196)
2025-08-18T10:47:52.2789224Z     at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.execute (AbstractJavadocMojo.java:1850)
2025-08-18T10:47:52.2789596Z     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
2025-08-18T10:47:52.2789892Z     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
2025-08-18T10:47:52.2790237Z     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
2025-08-18T10:47:52.2790518Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
2025-08-18T10:47:52.2790856Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
2025-08-18T10:47:52.2791144Z     at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
2025-08-18T10:47:52.2791414Z     at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
2025-08-18T10:47:52.2791794Z     at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
2025-08-18T10:47:52.2792074Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
2025-08-18T10:47:52.2792473Z     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
2025-08-18T10:47:52.2792789Z     at io.takari.maven.builder.smart.SmartBuilderImpl.buildProject (SmartBuilderImpl.java:206)
2025-08-18T10:47:52.2793176Z     at io.takari.maven.builder.smart.SmartBuilderImpl$ProjectBuildTask.run (SmartBuilderImpl.java:71)
2025-08-18T10:47:52.2793430Z     at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:545)
2025-08-18T10:47:52.2793628Z     at java.util.concurrent.FutureTask.run (FutureTask.java:328)
2025-08-18T10:47:52.2793912Z     at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1095)
2025-08-18T10:47:52.2794189Z     at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:619)
2025-08-18T10:47:52.2794331Z     at java.lang.Thread.run (Thread.java:1447)
```

--- a/src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java
+++ b/src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java
@@ -5394,8 +5394,8 @@ public abstract class AbstractJavadocMojo extends AbstractMojo {
                     logError("MavenInvocationException: " + e.getMessage(), e);
 
                     try {
-                        String invokerLogContent = new String(Files.readAllBytes(invokerLogFile.toPath()),
-                            StandardCharsets.UTF_8);
+                        String invokerLogContent =
+                                new String(Files.readAllBytes(invokerLogFile.toPath()), StandardCharsets.UTF_8);
                         // TODO: Why are we only interested in cases where the JVM won't start?
                         // probably we should throw an error in all cases
                         // [MJAVADOC-275][jdcasey] I changed the logic here to only throw an error WHEN
@@ -5403,8 +5403,7 @@ public abstract class AbstractJavadocMojo extends AbstractMojo {
                         if (invokerLogContent.contains(JavadocUtil.ERROR_INIT_VM)) {
                             throw new MavenReportException(e.getMessage(), e);
                         }
-                    }
-                    catch (IOException ex) {
+                    } catch (IOException ex) {
                         // ignore
                     }
                 } finally {

--- a/src/main/java/org/apache/maven/plugins/javadoc/JavadocUtil.java
+++ b/src/main/java/org/apache/maven/plugins/javadoc/JavadocUtil.java
@@ -827,12 +827,11 @@ public class JavadocUtil {
 
         if (result.getExitCode() != 0) {
             try {
-                String invokerLogContent = new String(Files.readAllBytes(invokerLog.toPath()),
-                    StandardCharsets.UTF_8);
+                String invokerLogContent = new String(Files.readAllBytes(invokerLog.toPath()), StandardCharsets.UTF_8);
 
                 // see DefaultMaven
                 if (!invokerLogContent.contains("Scanning for projects...")
-                    || invokerLogContent.contains(OutOfMemoryError.class.getName())) {
+                        || invokerLogContent.contains(OutOfMemoryError.class.getName())) {
                     if (log != null) {
                         log.error("Error occurred during initialization of VM, trying to use an empty MAVEN_OPTS...");
 
@@ -842,19 +841,18 @@ public class JavadocUtil {
                     }
                 }
             } catch (IOException e) {
-              // ignore
+                // ignore
             }
             result = invoke(log, invoker, request, invokerLog, goals, properties, "");
         }
 
         if (result.getExitCode() != 0) {
             try {
-                String invokerLogContent = new String(Files.readAllBytes(invokerLog.toPath()),
-                    StandardCharsets.UTF_8);
+                String invokerLogContent = new String(Files.readAllBytes(invokerLog.toPath()), StandardCharsets.UTF_8);
 
                 // see DefaultMaven
                 if (!invokerLogContent.contains("Scanning for projects...")
-                                || invokerLogContent.contains(OutOfMemoryError.class.getName())) {
+                        || invokerLogContent.contains(OutOfMemoryError.class.getName())) {
                     throw new MavenInvocationException(ERROR_INIT_VM);
                 }
 

